soe/soe.go

```diff
@@ -0,0 +1,61 @@
+package soe
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/hamba/avro/v2"
+)
+
+// Magic is the two-byte magic marker described in:
+// https://avro.apache.org/docs/1.10.2/spec.html#single_object_encoding
+var Magic = []byte{0xc3, 0x01}
+
+// AvroGenerated is implemented by all avrogen-generated Avro types.
+type AvroGenerated interface {
+	Unmarshal(b []byte) error
+	Marshal() ([]byte, error)
+	Schema() avro.Schema
+}
+
+// GetSchema returns the avro.Schema associated with type T.
+func GetSchema[T AvroGenerated]() avro.Schema {
+	// The Schema method returns a global schema, so can be called on a nil
+	// receiver.
+	var val T
+	return val.Schema()
+}
+
+// ComputeFingerprint returns an SOE-compatible (CRC64, little-endian) schema
+// fingerprint.
+func ComputeFingerprint(schema avro.Schema) ([]byte, error) {
+	return schema.FingerprintUsing(avro.CRC64AvroLE)
+}
+
+// ParseHeader validates SOE magic and splits data into fingerprint,rest.
+func ParseHeader(data []byte) ([]byte, []byte, error) {
```

Comment:
How many of these functions need to be exposed part of the package API? Seems odd that they are all external.

Reply:
Ah, yes. I'll look into that. They're kind of handy for testing, but not sure they all need to be exported.

Reply:
Never ended up doing anything with these. As I mentioned earlier, they're quite useful for the `soe` tests, but I think they can also help for tests and external code using `soe`. For example, `ComputeFingerprint` will typically be used by all resolver implementations.